### ISSUE TITLE
[wasm-mt] The System.Xml.Tests.TCValidateAfterAddInvalidSchema.AddValid_Import_Include_Redefine test gets stuck on Browser with multi-threading enabled

### ISSUE THREAD

**Issue by simonrozsival** [dotnet/runtime#75132]
The test fails processing the `schB8.xsd` input file. This input differs from the other test inputs by including a schema with a HTTP location:

```xml
<xsd:include schemaLocation="http://foo/foo" />
```

This is likely related to #75123

Ref #74407

**Comment by pavelsavara** [comment]
stack trace would help, @ilonatommy could you please try to reproduce this ?

**Comment by ilonatommy** [comment]
I know what the fix could be but I don't know how to do it neatly. We are using `s_defaultReaderSettings` here
https://github.com/dotnet/runtime/blob/ac56247d2e04d74d1e92a20b3c6c67fd3b368b8b/src/libraries/System.Private.Xml/src/System/Xml/Core/XmlReaderSettings.cs#L15
that works synchronically. We are able to do e.g.
```
#if FEATURE_WASM_THREADS
        internal static readonly XmlReaderSettings s_defaultReaderSettings = new() { Async = true }; // what about readOnly?
#else
        internal static readonly XmlReaderSettings s_defaultReaderSettings = new() { ReadOnly = true };
#endif
```
but as I pointed in the comment: we're doing it with the cost of ReadOnly functionality and I don't think it is an accepted soluton. I will make a PR to discuss it.

**Comment by pavelsavara** [comment]
In order to make those tests work, we need to have the tested file available in the VFS. 
Does this test run/work in ST build ?

**Comment by ilonatommy** [comment]
> Does this test run/work in ST build ?

No, it does not run, it's blocked for browser platform. When enabled, ST has the same stack trace as MT.

**Comment by pavelsavara** [comment]
See how `src\libraries\System.Runtime.Loader\tests\System.Runtime.Loader.Tests.csproj` is using `WasmFilesToIncludeInFileSystem` and do the same for `src\libraries\System.Private.Xml\tests\System.Private.Xml.Tests.csproj`

**Comment by ilonatommy** [comment]
In published files (also in `PublishItemsOutputGroupOutputs` values) we have test data from `src\libraries\System.Private.Xml\tests\Xslt\TestFiles` when we need test data from `src\libraries\System.Private.Xml\tests\XmlSchema\TestFiles`. I ma not sure why but this is the real reason for the issue.

**Comment by pavelsavara** [comment]
This could be OK now, let's re-test

**Comment by ilonatommy** [comment]
`AddValid_Import_Include_Redefine` is not blocked, `System.Private.Xml.Tests` is not excluded, so it's running on main and not failing.
